src/scraper/scrape.ts

```diff
@@ -6,9 +6,50 @@ import {
 import { AccountConfig } from "../types.js";
 import { ScraperErrorTypes } from "israeli-bank-scrapers/lib/scrapers/errors.js";
 import { createLogger } from "../utils/logger.js";
+import { config } from "../config.js";
+import { requestOtpCode } from "../bot/notifier.js";
 
 const logger = createLogger("scrape");
 
+/**
+ * Creates an OTP code retriever function for OneZero accounts
+ */
+function createOtpCodeRetriever(phoneNumber: string): () => Promise<string> {
+  return async () => {
+    if (!config.options.notifications.telegram?.enableOtp) {
+      throw new Error("OTP is not enabled in configuration");
+    }
+
+    logger(
+      `Requesting OTP code for phone number: ${phoneNumber.substring(0, 4)}...`,
+    );
+    return await requestOtpCode(phoneNumber);
```

Comment:
You only pass the phone number, but a user can have multiple accounts. Add the company ID and the account number so the user knows which otp code to send.